### DISCUSSION TITLE
how to simulate anaerobic growth

### DISCUSSION THREAD

**Discussion by Jungsday** [SysBioChalmers/yeast-GEM#299] [HELP!]
Hello, I tried to simulate anaerobic growth in this model.
So I modified glucose consumption and oxygen exchange simply by using the command below.

```matlab
model = changeRxnBounds(model,'r_1714',-18.5,'l');
model = changeRxnBounds(model,'r_1992',0,'l');
model = changeObjective(model,'r_2111');
FBAsolution = optimizeCbModel(model,'max')
fluxData = FBAsolution.v;
nonZeroFlag = 1;
printFluxVector(model, fluxData, nonZeroFlag)
```

However, I cant see the growth from the yeast-GEM_8.5 model.
What should I do for the anaerobic growth?

**Comment by edkerk** [comment]
If you run the following MATLAB code for the yeast-GEM repository root, this should give a functional model:

```matlab
initCobraToolbox(false)
cd ./code
model = loadYeastModel();
cd otherChanges\
modelAnaerobic = anaerobicModel(model);
modelAnaerobic = changeRxnBounds(modelAnaerobic,'r_1714',-18.5,'l');
modelAnaerobic = changeObjective(modelAnaerobic,'r_2111');
FBAsolution = optimizeCbModel(modelAnaerobic,'max');
printFluxVector(modelAnaerobic,FBAsolution.x,1)
```

A discussion on the arguments for these changes are available here: https://github.com/SysBioChalmers/yeast-GEM/issues/291.

**Reply by Jungsday** [reply]
Thanks.
I try the same thing but
When I run the code 'modelAnaerobic = anaerobicModel(model);'
There are errors. 
1) anaerobicModel>changeGAM (line 51)
bioPos = strcmp(model.rxnNames,'biomass pseudoreaction'); 
2) anaerobicModel (line 16)
model = changeGAM(model,GAM,NGAM);
Can you give me an advice for this?

Thank you.

**Reply by edkerk** [reply]
Can you please post the complete error message? I don't get any errors on my PC here.

**Reply by feiranl** [reply]
Maybe it is due to the name of the biomass reaction? Could you run this `model.rxnNames(strcmp(model.rxns,'r_4041'))`

**Reply by Jungsday** [reply]
It says

Struct contents reference from a non-struct array object.

Error in anaerobicModel>changeGAM (line 51)
bioPos = strcmp(model.rxnNames,'biomass pseudoreaction');

Error in anaerobicModel (line 16)
model = changeGAM(model,GAM,NGAM);

Whn I run model.rxnNames(strcmp(model.rxns,'r_4041')), 
The results was
ans =

  cell

    'biomass pseudoreaction'

**Reply by edkerk** [reply]
What is your MATLAB version and are you using the latest COBRA?

**Reply by Jungsday** [reply]
Except for the 1st change, when I input the rest as a command, growth was possible under anaerobic conditions.
Could you please tell me the unit of substrate the yeast consumes and the biomass it produces?
Is the consumption rate (mmol gDW-1 hr-1)/growth rate (hr-1)?

I really appreciate your help.

%1st change: Refit GAM and NGAM to exp. data, change biomass composition
GAM   = 30.49;  %Data from Nissen et al. 1997
P     = 0.461;  %Data from Nissen et al. 1997
NGAM  = 0;      %Refit done in Jouthen et al. 2012

model = changeGAM(model,GAM,NGAM);
model = scaleBioMass(model,'protein',P,'carbohydrate',false);

**Reply by feiranl** [reply]
Yes, the unit of the substrate consumption rate is mmol/gDW/h, and for biomass is h-1. It is quite strange that you cannot perform the 1st change. Did you try that one command by one command?

```
GAM = 30.49; %Data from Nissen et al. 1997
P = 0.461; %Data from Nissen et al. 1997
NGAM = 0; %Refit done in Jouthen et al. 2012

bioPos = strcmp(model.rxnNames,'biomass pseudoreaction');
for i = 1:length(model.mets)
    S_ix  = model.S(i,bioPos);
    isGAM = sum(strcmp({'ATP [cytoplasm]','ADP [cytoplasm]','H2O [cytoplasm]', ...
        'H+ [cytoplasm]','phosphate [cytoplasm]'},model.metNames{i})) == 1;
    if S_ix ~= 0 && isGAM
        model.S(i,bioPos) = sign(S_ix)*GAM;
    end
end
 pos = strcmp(model.rxnNames,'non-growth associated maintenance reaction');%NGAM
 model = setParam(model,'eq',model.rxns(pos),NGAM);% set both lb and ub to be mu
```

**Reply by jiuia** [reply]
@Jungsday hello, are you using the COBRA toolbox? And did you solve the problem?
